From 1f6f4e29769adb9d49a3c4831dd2ba61f6ccbdce Mon Sep 17 00:00:00 2001
From: Yogesh Tyagi <yogesh.tyagi@intel.com>
Date: Mon, 11 Mar 2024 13:13:32 +0800
Subject: [PATCH] Makefile: Fix build Issue
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In case build directory is different from source, make sure make is able to find the correct input files.

Fixes:
| dbus-binding-tool --prefix=thd_dbus_interface --mode=glib-server --output=thd_dbus_interface.h ../git/src/thd_dbus_interface.xml
| glib-compile-resources --generate-source thermald-resource.gresource.xml
| Failed to open file “thermald-resource.gresource.xml”: No such file or directory

Upstream-Status: Submitted
https://github.com/intel/thermal_daemon/pull/436

Signed-off-by: Yogesh Tyagi <yogesh.tyagi@intel.com>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index dd2ef10..cb19e17 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -97,6 +97,6 @@ thd_dbus_interface.h: $(top_srcdir)/src/thd_dbus_interface.xml
 	$(AM_V_GEN) dbus-binding-tool --prefix=thd_dbus_interface --mode=glib-server --output=$@ $<
 
 thermald-resource.c: $(top_srcdir)/thermald-resource.gresource.xml
-	$(AM_V_GEN) glib-compile-resources --generate-source thermald-resource.gresource.xml
+	$(AM_V_GEN) glib-compile-resources --generate-source --sourcedir=${top_srcdir} $<
 
 CLEANFILES = $(BUILT_SOURCES)
